packages/api/cms-api/src/user-permissions/user-permissions.module.ts

```diff
@@ -34,6 +49,10 @@ export class UserPermissionsModule {
                     provide: USER_PERMISSIONS_OPTIONS,
                     useValue: options,
                 },
+                {
+                    provide: ACCESS_CONTROL_SERVICE,
+                    useValue: options.accessControlService ?? new AccessControlService(),
```

Comment:
Shouldn't this be `useClass`? At least that's how services are provided: https://docs.nestjs.com/fundamentals/custom-providers#standard-providers

Reply:
Here it would work to require a class but in `forRootAsync` [useFactory](https://github.com/vivid-planet/comet/pull/1331/files#diff-87c5a6413e447c9c46e24a82309ca67e8200fd20332af148f99c70344145a287R65) is used which has to return the current instance.

The same applies for the [userService](https://github.com/vivid-planet/comet/pull/1331/files#diff-4ae73ee27568afd0d8888ef19b980d6894a319c0b76e635a94fa025b909a3e76R26).

Do you know a better solution?